climada_petals/hazard/tc_rainfield.py

```diff
@@ -65,200 +214,1552 @@ def set_from_tracks(self, *args, **kwargs):
+
+        The `category` attribute is set to the value of the `category`-attribute
+        of each of the given track data sets.
+
+        The `basin` attribute is set to the genesis basin for each event, which
+        is the first value of the `basin`-variable in each of the given track data sets.
+
+        Optionally, the time-dependent rain rates can be stored using the `store_rainrates`
+        function parameter (see below).
+
+        Currently, two models are supported to compute the rain rates: R-CLIPER and TCR. The
+        R-CLIPER model is documented in Tuleya et al. 2007. The TCR model was used by
+        Zhu et al. 2013 and Emanuel 2017 for the first time and is documented in detail in
+        Lu et al. 2018. This implementation of TCR includes improvements proposed in
+        Feldmann et al. 2019. TCR's accuracy is much higher than R-CLIPER's at the cost of
+        additional computational and data requirements.
+
+        When using the TCR model make sure that your TC track data includes the along-track
+        variables "t600" (temperature at 600 hPa) and "u850"/"v850" (wind speed at 850 hPa). Both
+        can be extracted from reanalysis or climate model outputs. For "t600", use the value at the
+        storm center. For "u850"/"v850", use the average over the 200-500 km annulus around the
+        storm center. If "u850"/"v850" is missing, this implementation sets the shear component of
+        the vertical velocity to 0. If "t600" is missing, the saturation specific humidity is set
+        to a universal estimate of 0.01 kg/kg. Both assumptions can have a large effect on the
+        results (see Lu et al. 2018).
+
+        The implementation of the R-CLIPER model currently does not allow modifications, so that
+        `model_kwargs` is ignored with `model="R-CLIPER"`. While the TCR model can be configured in
+        several ways, it is usually safe to go with the default settings. Here is the complete list
+        of `model_kwargs` and their meaning with `model="TCR"` (in alphabetical order):
+
+        c_drag_tif : Path or str, optional
+            Path to a GeoTIFF file containing gridded drag coefficients (bottom friction). If not
+            specified, an ERA5-based data set provided with CLIMADA is used. Default: None
+        e_precip : float, optional
+            Precipitation efficiency (unitless), the fraction of the vapor flux falling to the
+            surface as rainfall (Lu et al. 2018, eq. (14)). Note that we follow the MATLAB
+            reference implementation and use 0.5 as a default value instead of the 0.9 that was
+            proposed in Lu et al. 2018. Default: 0.5
+        elevation_tif : Path or str, optional
+            Path to a GeoTIFF file containing digital elevation model data (in m). If not
+            specified, a topography at 0.1 degree resolution provided with CLIMADA is used.
+            Default: None
+        matlab_ref_mode : bool, optional
```

Comment:
Why do we need this. If you are sure it's a bug, it's better if it's gone, no? I'd rather add in the docstring that this is the original implementation, with a bug fix, and explain what the bug fix is.

Reply:
I agree that bugs should just be fixed if we are sure that they are bugs. I updated the description of the parameter.

Background: When I originally implemented this, I found quite a few strange "issues" in the MATLAB code. And it took some time to write them down and get feedback from the original authors about these "issues". I didn't want to remove the issues unless the original authors definitely confirm that they are unintended ("bugs"). When I wrote this docstring I still didn't have feedback from the authors. Today, the situation is different: I know that some of the issues really were bugs. Those bugs are now fixed in my implementation and the original authors also decided to fix them in their implementation. But some of the "issues" were actually deliberate decisions by the original authors that I, personally, don't like (for example, using a parameter value that is not documented anywhere in the literature, not even in their own papers), and, at the same time, only make a comparably small difference. For testing purposes, I still think that it's useful to be able to replicate the behavior of the reference implementation. That's why I keep this parameter, but make clear what it actually does.